tsl/src/continuous_aggs/create.c

```diff
@@ -817,51 +823,54 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 					tbinfo->timezone = tz_name;
 					tbinfo->bucket_width = BUCKET_WIDTH_VARIABLE;
 				}
-				else
-				{
-					/*
-					 * Custom origin specified. This is always treated as
-					 * a variable-sized bucket case.
-					 */
-					tbinfo->bucket_width = BUCKET_WIDTH_VARIABLE;
+			}
+
+			if (list_length(fe->args) >= 5)
+				check_time_bucket_argument(lfourth(fe->args), "fifth");
```

Comment:
Should be `lfifth` I guess. There several other lines not covered by the tests, makes sense to add coverage for them as well.